shared_model/validators/field_validator.hpp

```diff
@@ -48,27 +48,37 @@ namespace shared_model {
           ReasonsGroupType &reason,
           const interface::types::AccountIdType &account_id) const {
         if (not std::regex_match(account_id, account_id_)) {
-          reason.second.push_back("Wrongly formed account_id");
+          auto message =
+              generateErrorMessage("Wrongly formed account_id",
+                                   static_cast<std::string>(account_id));
+          reason.second.push_back(std::move(message));
         }
       }
 
       void validateAssetId(
           ReasonsGroupType &reason,
           const interface::types::AssetIdType &asset_id) const {
         if (not std::regex_match(asset_id, asset_id_)) {
-          reason.second.push_back("Wrongly formed asset_id");
+          auto message = generateErrorMessage(
+              "Wrongly formed asset_id", static_cast<std::string>(asset_id));
+          reason.second.push_back(std::move(message));
         }
       }
 
       void validateAmount(ReasonsGroupType &reason,
                           const interface::Amount &amount) const {
-        /* put here any validations*/
+        if (amount.intValue() == 0) {
```

Comment:
It is unsigned, but I agree to change the condition to > 0 and do that for tx_counter as well because it is clearer and I had the same question when I saw this code for tx_counter.